### ISSUE TITLE
Feature request: Atomic Swap

### ISSUE THREAD

**Issue by anonymous108** [Samourai-Wallet/samourai-wallet-android#337]
Hello dear devs. Please forgive me if this already listing somewhere.

I am very interested in a wallet feature called Atomic Swaps, where a person can trade one crypto currency for a different one if the blockchain supports that protocol. From my understanding bitcoin does support the Atomic Swap, yes?

Is Atomic Swap feature planned for the Samourai Wallet? If no, please forgive me for asking why.

I read somewhere that Atomic Swap is on github already but I forget where. It was supposed to be written in the Go language. Maybe it's not to hard to integrate but of course would require much testing!

Many thanks for your time with reading this.

**Comment by SamouraiDev** [comment]
Samourai Wallet is a bitcoin-only and on-chain only wallet. We only handle bitcoin transactions and are not currently considering doing anything off of L1. Thanks for stopping by.

**Comment by anonymous108** [comment]
You are very welcome, thank you Samourai Dev.

I am always learning more.

After some reading I discover that Atomic Swapping may open up the Lightening(?) Network to exploitation by rational entities, who, because of the possible fluctuations in price between 2 different cryptos, will keep pestering the network to do trades, so they have options.

This is something I think about already when I try to understand the Atomic Swap process, but I believed the clever people had a solution for this problem.

Please forgive me but how do I learn more about "L1"? I'm sorry I don't know about this.

Again many thanks!

**Comment by Technifocal** [comment]
@anonymous108 L1 means "layer 1", I.E. the blockchain level. Lightning is currently layer 2 (or 3, if you count channel factories as layer 2).

**Comment by anonymous108** [comment]
Yes I understand perfectly now, many thanks Technifocal. I am watching a TV show called To the Moon, they talk about the new layers like this. I should be guessing this! I am confused by one area on this show because last week they explained that the users rejected Segregated Witness in mid 2017, with UASF, but to my mind this is not true, because I hear about SegWit many times since then. To me,  its like everyone wants to use Segwit. Did high fees in Dec 2017 make users want the change?

If you can point me to a strong knowledge base on this area it would give me great joy. Only for my learning.

Thank You again Technifocal and SamouraiDev
